### ISSUE TITLE
S3 upload fails on large files over 550+ Mb

### ISSUE THREAD

**Issue by avishnyakov** [localstack/localstack#938]
Hey guys, 

Looking into S3 mocks capabilities, they work very nice so far.
That said, having failures with large files upload - anything more that 550+ Mb seems to be failing.

Using both aws s3 cp and aws s3 sync - same-same result.
Looks that localstack processes within the container get killed, going to look into more debug/details around this. Outside looks as this:

```
ktcn-localstack_1  | 2018-09-22T04:14:45:ERROR:localstack.services.generic_proxy: Error forwarding request: HTTPConnectionPool(host='127.0.0.1', port=4563): Max retries exceeded with url: /artefact-store (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f84feb7c190>: Failed to establish a new connection: [Errno 111] Connection refused',)) Traceback (most recent call last):
```

Meanwhile, just wonder if anyone managed to get large file upload working or if there is additional setup to be done?

**Comment by avishnyakov** [comment]
Turned out that 1Gb swap space wasn't enough. Upload would be crashing due to maxed out swap. Tuned containers to add more swap/ram, all works fine.

**Comment by kprzerwa-r7** [comment]
I managed to sort out similar issue by increasing read timeout.
Using ruby aws sdk:
```
Aws.config[:s3] = {
      http_read_timeout: 120
}
```
In `aws-cli`:
```
awslocal s3api put-object --cli-read-timeout 0 --bucket <bucket> --key <key> --body <path>
```

**Comment by kprzerwa-r7** [comment]
Note that `--cli-read-timeout 0` switches the timeout completely

**Comment by avishnyakov** [comment]
Interesting, @kprzerwa-r7. In my case we were simply running out of ram. 
I can't avoid noticing that this was open since  Sep 22, 2018 with no comments from the team :(

**Comment by whummer** [comment]
Thanks for reporting @avishnyakov @kprzerwa-r7 - this should be fixed in #1695 . Closing this issue, please re-open if the problem persists. Thanks
